docs/en/manuals/extensions_best_practices.md

```diff
@@ -0,0 +1,131 @@
+# Best Practices
+
+Writing cross platform code can be difficult, but there are some ways to make it easier to both develop and maintain.
+Here we list some ways we at Defold work with cross platform native code and API's.
+
+## Defold code
+
+In the Defold engine we use C++ very sparingly. In fact, most code is very C-like.
+We avoid templates, except for a few container classes, due to the fact that templates both incurs a cost on compilation times
+as well as executable size.
+
+### C++ version
+
+The defold source it built with the default C++ version of each compiler (See [Native Extensions - Best Practices](/manuals/extensions_best_practices#_)).
+
+We avoid using the latest features or versions of C++. Mostly because we already have what we need to build an game engine.
+Keeping track of the latest features of C++ is a time consuming task, and to really master those features will require a lot of precious time.
+
+It also has the added benefit for our extension developers that we keep a stable ABI.
+
```

Comment:
Perhaps also mentioning that using the latest features may make it harder to compile on different platforms due to varying support?